### PR TITLE
Fix RTL currency symbols not respecting the currency position setting in the React Checkout blocks

### LINKED ISSUE #67688: Honor the currency position setting for RTL-script symbols in block-rendered prices (FormattedMonetaryAmount)
Follow-up to WOOPLUG-559 / [woo#31637](<https://github.com/woocommerce/woocommerce/issues/31637>). [PR #67686](<https://github.com/woocommerce/woocommerce/pull/67686>) fixes the PHP side by adding `dir="auto"` to the currency symbol span in `wc_price()`; every server-rendered price (classic templates, widgets, Product Price block, emails) now honors the currency position setting for RTL-script symbols such as the Lebanese Pound (`ل.ل`).

Block-rendered prices are not covered. The Cart, Checkout, and Mini-Cart blocks format prices client-side in `FormattedMonetaryAmount` (`plugins/woocommerce/client/blocks/packages/public-api/blocks-components/formatted-monetary-amount/index.tsx`), which composes a single plain-text string (`prefix + amount + suffix` via `react-number-format`) with no markup around the symbol and no bidi isolation; the Unicode bidi algorithm reorders RTL-script symbols regardless of the configured position.

**Proposed fix**

Wrap the symbol inside the prefix/suffix strings in first-strong isolate characters (U+2068 FSI … U+2069 PDI); the plain-text equivalent of `dir="auto"`. Constraints:

* Apply to display mode only; `FormattedMonetaryAmount` is also used for editable inputs (price filter), and invisible control characters must not leak into input values.
* Test on genuinely RTL sites (Hebrew/Arabic locales). A previous CSS attempt for the Product Price block ([PR #58134](<https://github.com/woocommerce/woocommerce/pull/58134>), used `unicode-bidi: bidi-override`) reversed digits on all RTL sites ([woo#60986](<https://github.com/woocommerce/woocommerce/issues/60986>)) and was reverted in [PR #61003](<https://github.com/woocommerce/woocommerce/pull/61003>). Isolates cannot reorder characters within a run, but the regression scenario must be covered by tests.

Related: WOOPLUG-3255 (Won't Do) asked for a separate HTML element for the currency symbol in this component; isolate characters achieve the bidi goal without changing the markup contract.

### PR DESCRIPTION
**Pull request by ralucaStan**
### Submission Review Guidelines:

-   [x] I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   [x] I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

Closes #67688.
Closes WOOPLUG-7516
Block prices compose the currency symbol and the amount into one flat string, so the bidi algorithm is free to reorder them. A currency with an RTL-script symbol (Lebanese pound `ل.ل`, Shekel, Dirham) ignores the currency position setting completely: a store set to "Left with space" renders `1,563.45 ل.ل`.

This PR aligns block prices with the same markup `wc_price()` produces.

Depends on #67686 for the same fix on the PHP side. Mini-Cart handled separately in WOOPLUG-7536

### Screenshots or screen recordings:

Lebanese pound, position "Left with space", English store. `SYM` stands in for `ل.ل`; printing the real symbol here would be reordered by the browser, which is the bug itself.

| | Rendered |
| --- | --- |
| Before | `1,563.45 SYM` |
| After | `SYM 1,563.45` |

### How to test the changes in this Pull Request:

1. **WooCommerce > Settings > General**: set **Currency** to **Lebanese pound (ل.ل)** and **Currency position** to **Left with space**. Keep the site language English.
2. Add a product to the cart and open the Cart block page; `ل.ل` should sit on the **left** of every amount, including a discounted line item.
3. Enable **display prices including tax** and **itemised taxes**; the "Including …" line under the cart total should place the symbol on the configured side too.
4. Check Checkout the checkout block as well

### Testing that has already taken place:

Manual testing on a Studio store.

**Backward compatibility:** `FormattedMonetaryAmount` is public API, exported from `@woocommerce/blocks-components` (`wc.blocksComponents`). No prop changes. The output gains a `<bdi>` and a classed symbol span;

### Milestone

- [x] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**

### Changelog entry

-   [x] This Pull Request contains a changelog entry created manually (`plugins/woocommerce/changelog/fix-wooplug-7516-rtl-currency-symbol-isolation`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by ralucaStan** [comment]
## Matrix tested on a real store

Re-run on Aug 18 against 9b027b7ff (the branch with the review changes applied), on a fresh Studio site running this build.

Block prices honor the setting in all 16 combinations. Every ❌ is PHP, and #67686 closes those.

One page renders both paths side by side: the `[products]` shortcode (PHP, `wc_price()`) and the All Products block (React, `FormattedMonetaryAmount`). The PHP column is this branch without #67686, so it matches trunk. The tables give the side the symbol is **drawn** on, measured from the rendered geometry (`getBoundingClientRect` of the symbol against its `<bdi>`), not the string; printing the symbol here would be reordered by the browser, which is the bug itself.

### English site (LTR)

| Currency | Position | Expected | PHP | Blocks |
| --- | --- | --- | --- | --- |
| LBP | Left | left | right ❌ | left ✅ |
| LBP | Left with space | left | right ❌ | left ✅ |
| LBP | Right | right | left ❌ | right ✅ |
| LBP | Right with space | right | left ❌ | right ✅ |
| USD | Left | left | left ✅ | left ✅ |
| USD | Left with space | left | left ✅ | left ✅ |
| USD | Right | right | right ✅ | right ✅ |
| USD | Right with space | right | right ✅ | right ✅ |

### Arabic site (RTL)

| Currency | Position | Expected | PHP | Blocks |
| --- | --- | --- | --- | --- |
| LBP | Left | left | right ❌ | left ✅ |
| LBP | Left with space | left | right ❌ | left ✅ |
| LBP | Right | right | left ❌ | right ✅ |
| LBP | Right with space | right | left ❌ | right ✅ |
| USD | Left | left | left ✅ | left ✅ |
| USD | Left with space | left | left ✅ | left ✅ |
| USD | Right | right | right ✅ | right ✅ |
| USD | Right with space | right | right ✅ | right ✅ |

A note for anyone repeating this: after changing the currency settings, two caches can keep serving the old side. Run `wp cache flush` for the options cache, and hard-refresh the Store API responses in the browser; the products collection response is HTTP-cached, so a plain reload renders the previous currency.

File: plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/order-summary/test/index.js [hunk 1]
```diff
@@ -9,6 +9,15 @@ import { previewCart } from '@woocommerce/resource-previews';
  */
 import OrderSummary from '../index';
 
+// The currency symbol sits in its own element, so the whole price is read at once.
+/**
+ * @param {HTMLElement} container
+ */
+const getLineSubtotal = ( container ) =>
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/order-summary/test/index.js:16]: If I understand correctly, I don't think this is necessary? We can still assert the correct value with `getByText` without adding a new function and operating on the container I think.

File: plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/shipping-rates-control-package/test/index.tsx [hunk 2]
```diff
@@ -59,16 +59,23 @@ test( 'renders available shipping rates', async () => {
 		/>
 	);
 
+	// The space between the symbol and the amount is a jsdom artifact: its
+	// accessible-name computation joins text across element boundaries with a
+	// space, and the currency symbol sits in its own element now. Real
+	// browsers don't insert it — Chrome's accessibility tree reports
+	// "Flat rate $10.00".
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/shipping-rates-control-package/test/index.tsx:66]: Lets remove this. Or just leave `The space between the symbol and the amount is a jsdom artifact`

File: plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/test/index.tsx [hunk 3]
```diff
@@ -13,6 +13,11 @@ import { previewCart as mockPreviewCart } from '@woocommerce/resource-previews';
  */
 import TotalsFooterItem from '../index';
 
+// The currency symbol sits in its own element, so the whole value is read at once.
+const getTotalValue = ( container: HTMLElement ) =>
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/test/index.tsx:17]: Can we not just assert the text of the container (which contains the price) instead of another function querying a sub-elements text content for just the price?

File: plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/test/index.tsx [hunk 4]
```diff
@@ -270,4 +277,41 @@ describe( 'TotalsFooterItem', () => {
...
+			'Including £0.50 10% VAT, £0.50 5% VAT'
+		);
+
+		// Each amount is a price element with the symbol in its own isolate,
+		// so the bidi algorithm cannot move it to the other side of the digits.
+		const amounts = taxInfo?.querySelectorAll( 'bdi' );
+		expect( amounts ).toHaveLength( 2 );
+		amounts?.forEach( ( amount ) => {
+			expect( amount.textContent ).toBe( '£0.50' );
+			expect(
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/test/index.tsx:310]: Do we need this if we have asserted the above? Feels like the above assertion covers this.

File: plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/test/index.tsx [hunk 5]
```diff
@@ -270,4 +277,41 @@ describe( 'TotalsFooterItem', () => {
...
+		const { container } = render(
+			<TotalsFooterItem currency={ currency } values={ valuesWithTax } />
+		);
+
+		const taxInfo = container.querySelector(
+			'.wc-block-components-totals-footer-item-tax'
+		);
+
+		// The line reads the same as it did when it was one flat string.
+		expect( taxInfo?.textContent ).toBe(
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/test/index.tsx:300]: I am pretty sure we can replace querySelector usage with [textContentMatcher](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/blocks/tests/utils/find-by-text.ts#L10) and/or RTL native APIs here which would be better.

File: plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/index.tsx [hunk 6]
```diff
@@ -111,19 +110,31 @@ const TotalsFooterItem = ( {
 
 	const parsedTaxValue = parseInt( totalTax, 10 );
 
+	// Each amount is a component rather than a formatted string so that the
+	// currency symbol keeps its isolation; a flat string lets the bidi
+	// algorithm move an RTL symbol to the wrong side of the amount.
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/index.tsx:115]: Lets remove this. I don't think its necessary and tests cover it.

File: plugins/woocommerce/client/blocks/assets/js/base/components/product-price/test/index.js [hunk 7]
```diff
@@ -36,6 +36,15 @@ describe( 'ProductPrice', () => {
 		thousandSeparator: ',',
 	};
 
+	// The currency symbol sits in its own element, so each price is read whole.
+	const getRegularPrice = ( container ) =>
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/assets/js/base/components/product-price/test/index.js:40]: Just echoing my comments above, I dont think we need to create custom functions for all these tests

File: plugins/woocommerce/client/blocks/packages/public-api/blocks-components/formatted-monetary-amount/style.scss [hunk 8]
```diff
@@ -1,3 +1,9 @@
 .wc-block-components-formatted-money-amount {
 	white-space: nowrap;
 }
+
+.wc-block-components-formatted-money-amount__currency-symbol {
+	// `dir` only isolates through the UA rule `[dir] { unicode-bidi: isolate }`,
+	// which a theme reset can undo. Pin it so the symbol stays in place.
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/packages/public-api/blocks-components/formatted-monetary-amount/style.scss:7]: Lets remove

File: plugins/woocommerce/client/blocks/packages/public-api/blocks-components/formatted-monetary-amount/index.tsx [hunk 9]
```diff
@@ -106,15 +196,34 @@ const FormattedMonetaryAmount = ( {
 		className
 	);
 	const decimalScale = props.decimalScale ?? currency?.minorUnit;
+	const decodedPrefix = decodeHtmlEntities( currency.prefix );
+	const decodedSuffix = decodeHtmlEntities( currency.suffix );
+
+	// Compose the price as markup so it matches the structure `wc_price()`
+	// produces everywhere else in the store. An input value cannot hold
+	// elements, and a consumer-supplied renderText owns its output, so both keep
+	// the plain string.
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/packages/public-api/blocks-components/formatted-monetary-amount/index.tsx:205]: Feel like we should remove this too, unless you think this information is important.

File: plugins/woocommerce/client/blocks/assets/js/base/components/product-price/test/index.js [hunk 10]
```diff
@@ -58,6 +68,93 @@ describe( 'ProductPrice', () => {
...
+				regularPrice={ 100 }
+				currency={ currency }
+			/>
+		);
+
+		expect( container.querySelector( 'del' )?.textContent ).toBe( '£1.00' );
+		expect( container.querySelector( 'ins' )?.textContent ).toBe( '£0.50' );
+	} );
+
+	test( 'isolates the currency symbol of both sale prices', () => {
```
Comment (tjcafferkey) [inline plugins/woocommerce/client/blocks/assets/js/base/components/product-price/test/index.js:131]: This looks like a duplicate test of one above which uses `getRegularPrice` and `getDiscountPrice`?

**Review by tjcafferkey — CHANGES_REQUESTED** [review]
The production code works as expected @ralucaStan so thanks for fixing 🎉 ! But I feel like there's some funky test changes here. I have noticed and comment on some, but there could be more. There appears to be some patterns though with what I've seen:

1. We are using DOM APIs in tests to bypass React Testing Library APIs, this breaks the pattern and purpose of using React Testing Library
2. We are adding test cases that are either duplicate or covered by other tests already

I think it would be good to review these tests before merging, and ensuring that we are testing correctly, but also only included tests that we deem useful or necessary.

File: plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/test/index.tsx (hunk 5 above)
Comment (ralucaStan) [reply plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/test/index.tsx]: done in c93008a7bcdfd6ef3a0299c8886520fb5ad47aa8

File: plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/index.tsx (hunk 6 above)
Comment (ralucaStan) [reply plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/totals/footer-item/index.tsx]: f5992235ca4e462f6d46cc318dbdb02570ff0d00

File: plugins/woocommerce/client/blocks/packages/public-api/blocks-components/formatted-monetary-amount/style.scss (hunk 8 above)
Comment (ralucaStan) [reply plugins/woocommerce/client/blocks/packages/public-api/blocks-components/formatted-monetary-amount/style.scss]: I'd like to keep this one as it's and essential line for the prices to work. renamed in 23b20b8836f13a3511570118553a8150cb09c55a

File: plugins/woocommerce/client/blocks/assets/js/base/components/product-price/test/index.js (hunk 10 above)
Comment (ralucaStan) [reply plugins/woocommerce/client/blocks/assets/js/base/components/product-price/test/index.js]: 2d99d266404e1211719ed12a26a45a4f9ad85f39

**Comment by ralucaStan** [comment]
@tjcafferkey thank you for the attention to tests, they should be in a better shape. This is ready for another review

**Review by tjcafferkey — APPROVED** [review]
Thanks @ralucaStan this now LGTM - there is a failing job which might need looking at but approving in the meantime
